react/Gallery.js

```diff
@@ -1,34 +1,104 @@
 import React from 'react'
-import { PropTypes } from 'prop-types'
 import classNames from 'classnames'
+import { List, WindowScroller, AutoSizer, CellMeasurer, CellMeasurerCache } from 'react-virtualized'
+import { PropTypes } from 'prop-types'
+
+import { withRuntimeContext, NoSSR } from 'render'
 
 import { productShape } from './constants/propTypes'
 import GalleryItem from './components/GalleryItem'
 
 /**
  * Canonical gallery that displays a list of given products.
  */
-const Gallery = ({ products, summary, layoutMode }) => {
-  const classes = classNames('vtex-gallery pa3 bn', {
-    'vtex-gallery--two-columns': layoutMode === 'small',
+
+const Gallery = ({
+  products,
+  summary,
+  layoutMode,
+  runtime: { hints: { mobile } },
+  itemWidth }) => {
+  const cache = new CellMeasurerCache({
+    defaultHeight: 305,
+    fixedWidth: true,
+    keyMapper: () => 1,
   })
 
+  const renderRow = ({ index, key, style, parent, itemsPerRow }) => {
+    const from = index * itemsPerRow
+    const rowItems = products.slice(from, from + itemsPerRow)
+
+    const containerClasses = classNames('Row vtex-gallery__row', {
```

Comment:
Why this class `Row`?

Reply:
I thought it was needed to react virtualized identify the rows, but it isn't, thanks for asking.